oci-image/src/snapshots/mod.rs

```diff
@@ -0,0 +1,39 @@
+pub mod overlay;
+
+use crate::Result;
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum SnapshotType {
+    Overlay,
+}
+
+impl ToString for SnapshotType {
+    fn to_string(&self) -> String {
+        String::from(match self {
+            SnapshotType::Overlay => "overlay",
+        })
+    }
+}
+
+/// The `Snapshotter` trait defines methods that can be implemented in order to create a container image snapshot.
+pub(crate) trait Snapshotter: Send + Sync {
+    fn mount(&mut self, layers: Vec<String>, mount_path: &Path) -> Result<MountPoint>;
```

Comment:
As commented on the ImageManager and State commits, you should pass the snapshot_index to the mount method.